commands/base_commands/rolling.py

```diff
@@ -68,65 +69,143 @@ class CmdDiceCheck(ArxCommand):
+        """
+        split_str = args.split("|")
+        if not split_str[0].lstrip('-').isdigit():
+            raise ValueError("Usage: @check/retainer <id>|<stat>[+<skill>][ at <difficulty number>][=receiver1,receiver2,etc]")
+
+        if split_str[0].lstrip('-').isdigit() and not int(split_str[0]) > 0:
+            raise ValueError("Retainer ID must be a positive number.")
+
+        retainer_id = int(split_str[0])
+
+        return split_str[1], retainer_id
+
+
+    def _extract_difficulty(self, args: str) -> (str, int):
+        """
+        Extracts difficulty value if specified.  Otherwise, sets difficulty
+        to be DIFF_DEFAULT.
+
+        Args must be of the form '<stat>[+<skill>][ at <difficulty number>]'
+        
+        Returns difficulty and '<stat>[+<skill>]'
+        """
         difficulty = stats_and_skills.DIFF_DEFAULT
-        if len(diff_list) > 1:
-            if not diff_list[1].isdigit() or not 0 < int(diff_list[1]) < maximum_difference:
-                caller.msg("Difficulty must be a number between 1 and %s." % maximum_difference)
+        maximum_difference = 100
+
+        # if args contains ' at ', then we split into halves. otherwise it's default difficulty
+        split_str = args.split(' at ')
+        if len(split_str) > 1:
+            if not split_str[1].isdigit() or not 0 < int(split_str[1]) < maximum_difference:
+                raise ValueError("Difficulty must be a number between 1 and %s." % maximum_difference)
                 return
-            difficulty = int(diff_list[1])
-        args = diff_list[0]
-        arg_list = args.split("+")
-        if len(arg_list) > 1:
-            skill = arg_list[1].strip()
-        stat = arg_list[0].strip()
+            difficulty = int(split_str[1])
+        
+        return split_str[0], difficulty
+
+
+    def _extract_stat_skill(self, args: str) -> (str, str):
+        """
+        Extracts and validates the stat and skill names.
+
+        Args must be of the form '<stat>[+<skill>]'
+        """
+        stat, skill = None, None
+
+        split_str = args.split("+")
+        stat = split_str[0].strip()
+        if len(split_str) > 1:
+            skill = split_str[1].strip()
+
         matches = stats_and_skills.get_partial_match(stat, "stat")
         if not matches or len(matches) > 1:
-            caller.msg("There must be one unique match for a character stat. Please check spelling and try again.")
-            return
+            raise ValueError("There must be one unique match for a character stat. Please check spelling and try again.")
         # get unique string that matches stat
         stat = matches[0]
+
         if skill:
             matches = stats_and_skills.get_partial_match(skill, "skill")
             if not matches:
                 # check for a skill not in the normal valid list
-                if skill in caller.db.skills:
+                if skill in self.caller.db.skills:
```

Comment:
`.db.skills` is deprecated. If you could, please use the `traitshandler` for checking skills